### ISSUE TITLE
OS X 14.0 unexpected fault address when debugging hello world

### ISSUE THREAD

**Issue by suntehnik** [go-delve/delve#3519]
Please answer the following before submitting your issue:

Note: Please include any substantial examples (debug session output,
stacktraces, etc) as linked gists.

If this is about source listing not showing up (or breakpoints not being
accepted) while running in a container please read our
[FAQ](https://github.com/go-delve/delve/blob/master/Documentation/faq.md#substpath)
first.

1. What version of Delve are you using (`dlv version`)?
`
dlv version
Delve Debugger
Version: 1.21.1
Build: $Id: a358c02f24aa7047ecc562b0587dc2d08330b2cf $
`
3. What version of Go are you using? (`go version`)?
`
go version
go version go1.21.1 darwin/amd64
`
5. What operating system and processor architecture are you using?
`
uname -a
Darwin XYZ.local 23.0.0 Darwin Kernel Version 23.0.0: Fri Sep 15 14:42:42 PDT 2023; root:xnu-10002.1.13~1/RELEASE_X86_64 x86_64
`
OS X 14.0, x86_64
7. What did you do?
Trying to debug "Hello, world!" app, put a breakpoint on Printf line.

`package main

import "fmt"

func main() {
	fmt.Printf("Hello, world!\n")
}`

9. What did you expect to see?
App should launch and stop on the breakpoint
10. What did you see instead?
I see error in debug console
`
unexpected fault address 0x10b0360
`

Removing or disabling breakpoint leads to successfull app launch.

Dalve log can be found [here](https://gist.github.com/suntehnik/d77f0db258378225c0562b75427b6536).

Error can be found [here](https://gist.github.com/suntehnik/d77f0db258378225c0562b75427b6536#file-gistfile1-txt-L547).

**Comment by derekparker** [comment]
Thanks for reporting this! 

I haven't yet updated my person machine to OS X 14.0 but plan to later this week, so I will investigate this early next week.

Out of curiosity, does this reproduce through the terminal interface? (e.g. simply running `dlv debug`, setting a breakpoint and continuing).

**Comment by suntehnik** [comment]
Here is the [gist](https://gist.github.com/suntehnik/9fecb6c00d0ec7125731783270a52431) with session log
TLDR; reproduceable with cli

**Comment by aarzilli** [comment]
Can I see a log with `--log --log-output=gdbwire`?

**Comment by suntehnik** [comment]
> Can I see a log with `--log --log-output=gdbwire`?

Dalve log can be found [here](https://gist.github.com/suntehnik/d77f0db258378225c0562b75427b6536).

**Comment by aarzilli** [comment]
It doesn't look like we are doing anything wrong and debugserver is crashing. Have you tried updating the command line tools like @dylanrayboss says?

**Comment by aarzilli** [comment]
Closing, there's probably nothing else to do about this.

**Comment by LionheartLann** [comment]
After several attempts, I found that reinstalling the command line tools did not resolve the problem. However, I'm pleased to report that downgrading the Go version from go1.22.4 to go1.19 successfully resolved the issue I was facing. This workaround might be helpful for others experiencing similar problems.

> environment
Delve Debugger Version: 1.22.1
go version go1.19 darwin/arm64
macos sonoma 14.4.1 amd64
goland 2024.1.4
